File: pascals-triangle/pascals-triangle.java
class Solution {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> tree = new ArrayList<>();
        List<Integer> leaf = new ArrayList<>();
        for (int i = 0; i < numRows; i++) {
            int prev = 1;
            leaf.add(1);
            for (int j = 1; i > 1 && j < i;j++) {
                int curr = leaf.get(j);
                leaf.set(j, prev + curr);
                prev = curr;
            }

            tree.add(new ArrayList<>(leaf));
        }

        return tree;
    }
}

File: pascals-triangle/pascals-triangle.go
func generate(numRows int) [][]int {
	tree := make([][]int, 0)
	leaf := make([]int, 0)
	for i := 0; i < numRows; i++ {
		prev := 1
		leaf = append(leaf, 1)
		for j := 1; i > 1 && j < i; j++ {
			curr := leaf[j]
			leaf[j] += prev
			prev = curr
		}

		copyLeaf := make([]int, len(leaf))
		copy(copyLeaf, leaf)
		tree = append(tree, copyLeaf)
	}

	return tree
}